pkg/build/reaper/reaper.go

```diff
@@ -0,0 +1,74 @@
+package reaper
+
+import (
+	"fmt"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/kubectl"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+)
+
+// NewBuildConfigReaper returns a new reaper for buildConfigs
+func NewBuildConfigReaper(oc *client.Client) kubectl.Reaper {
+	return &BuildConfigReaper{oc: oc, pollInterval: kubectl.Interval, timeout: kubectl.Timeout}
+}
+
+// BuildConfigReaper implements the Reaper interface for buildConfigs
+type BuildConfigReaper struct {
+	oc                    client.Interface
+	pollInterval, timeout time.Duration
+}
+
+// Stop deletes the build configuration and all of the associated builds.
+func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
+	// If the config is already deleted, it may still have associated
+	// builds which didn't get cleaned up during prior calls to Stop. If
+	// the config can't be found, still make an attempt to clean up the
+	// builds.
+	//
+	// We delete the config first to ensure no further builds will be created.
+	err := reaper.oc.BuildConfigs(namespace).Delete(name)
+	notFound := kerrors.IsNotFound(err)
```

Comment:
@deads2k and @smarterclayton convinced me this is not right (for jobs) and since there are no races with the BC controller I would vote for moving the bc deletion in the end. Hopefully, making the BC controller aware of a bc that is about to be deleted won't be difficult?